### PR TITLE
Clover

### PR DESCRIPTION
**Pull request by brqgoo**
# Grant Application Checklist

- [x] The [application-template.md](https://github.com/w3f/Open-Grants-Program/blob/master/applications/application-template.md) has been copied, renamed ( "project_name.md") and updated.
- [x] A BTC address for the payment of the milestones is provided inside the application.  
- [x] The software of the project will be released under the Apache license version 2.0 as specified in the [terms and conditions](https://gist.github.com/Noc2/75bc58e8ce9b5d419ff883b0cf2b8c19).
- [x] The total funding amount of the project is below CHF 30k at the time of submission.

### CODE REVIEW

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the application. Looks interesting. Seems you are working on a lot of things at the same time. Could you integrate the open source parts which you develop into the milestone delivery template https://github.com/w3f/Open-Grants-Program/blob/master/applications/application-template.md#milestone-1-example--implement-substrate-modules? The deliveries need to be part of the PR, because you can change the content of the link at any time.

**Comment by brqgoo** [comment]
Hi David, made respective changes accordingly. Let me know if anything else is missing. Thanks.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Sorry for the late response and thanks for the update. Somehow I missed that you updated the PR again. You need to resign the terms and conditions again, since we recently updated them. Also could you update the total costs (BTC or DAI amount)?  Additionally could you specify the programming language/framework for each deliverable?  And you can remove the following deliverables, since these aren’t any deliverables that are beneficial for the entire ecosystem, but rather are specific to your project: Clover Mainnet, Developer Incentive Program, Roadmap and Stress Test.

**Comment by brqgoo** [comment]
Hi @Noc2 , all sorted now.

File: applications/clover_network.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+* [Barek Sekandari](https://www.linkedin.com/in/barek-sekandari-597752153/)
+
+* [Marina Danylyuk](https://www.linkedin.com/in/marinadanylyuk/)
+
+## Development Roadmap :nut_and_bolt: 
+
+### Overview
+* **Total Estimated Duration:** 3 months
+* **Full-time equivalent (FTE):**  10 FTE
+* **Total Costs:** Amount of Payment in BTC for the whole project. The total amount of funding needs to be below $30k at the time of submission. (e.g. 0.80 BTC)
```
Comment (Noc2) [inline applications/clover_network.md:122]: Could you update this to show the complete total costs? (0.66 BTC)

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the update. Just one more thing:

File: applications/clover_network.md (hunk 1 above)
Comment (brqgoo) [reply applications/clover_network.md]: Just updated. Thanks.

**Comment by Noc2** [comment]
Thanks for the update. I will share it with the rest of the team.

**Review by Noc2 — APPROVED** [review]

**Comment by mmagician** [comment]
Could you be more specific about the following points & provide explanations and further information please?
1. > Fully implementing distributing gas fee functionality for EVM contract deployers
2. > Implementing enterprise developers' achitecture with in the low level logic
3. > Rolling out minor EVM adjustments with Clover chain
4. > Creating first alpha build for Clover cross-chain block explorer based on Rosetta protocol

File: applications/clover_network.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  We are a team of like-minded people who have a dream of building a groundbreaking technology towards censorship resistance on utilizing newest additions to DLT technologies. Apart from incentivizing our core team with a structured/periodic compensation plan throughout the next 5 years with cliffs imposed in between, we give ourselves a high degree of autonomy and lots of space for creativity overall.
+
+### Project Details 
+
+#### Mockups/designs  
+[https://www.figma.com/file/SWg7LONt81ot05S1nlszcH/Clover-Wallet?node-id=0%3A1](https://www.figma.com/file/SWg7LONt81ot05S1nlszcH/Clover-Wallet?node-id=0%3A1)
+
+
+#### API specs
+[https://docs.google.com/document/d/1UfNWe82X76a1LoKrSwSXCJV9gbcbHT6QLtsU8DIYWzY/edit?usp=sharing](https://docs.google.com/document/d/1UfNWe82X76a1LoKrSwSXCJV9gbcbHT6QLtsU8DIYWzY/edit?usp=sharing)
```
Comment (laboon) [inline applications/clover_network.md:38]: This appears to be a test run. Can you point to the actual API specs?

File: applications/clover_network.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+#### Team Interest
+
+  We are a team of like-minded people who have a dream of building a groundbreaking technology towards censorship resistance on utilizing newest additions to DLT technologies. Apart from incentivizing our core team with a structured/periodic compensation plan throughout the next 5 years with cliffs imposed in between, we give ourselves a high degree of autonomy and lots of space for creativity overall.
+
+### Project Details 
+
+#### Mockups/designs  
+[https://www.figma.com/file/SWg7LONt81ot05S1nlszcH/Clover-Wallet?node-id=0%3A1](https://www.figma.com/file/SWg7LONt81ot05S1nlszcH/Clover-Wallet?node-id=0%3A1)
```
Comment (laboon) [inline applications/clover_network.md:34]: It seems like these mockups only show basic wallet functionality. Do you have any showing any of the other planned functionality?

File: applications/clover_network.md [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+
+#### Documentation of core components, protocols, architecture
```
Comment (laboon) [inline applications/clover_network.md:44]: Of the core layer and UI, what is actually already completed?

File: applications/clover_network.md [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+| ------------- | ------------- | ------------- | ------------- |
+| 1. | Documentation | Text | Documentation will be ready at 1.0 version to cover 100% of the current features |
+| 2. | Tulip Editor | Vue/NodeJS/Solidity | Delivering alpha build for Tulip Editor with EVM support |
+| 3. | Clover Wallet App | Android/iOS Native | Releasing Clover wallet app on Google Play and Apple Store |  
+| 4. | Clover OS | Native/JS | Having Clover OS support for iOS/Android/Embedded by the end of Mar. |  
+| 5. | Clover Store | Vue/IPFS/Solidity/NodeJS | Delivering the official release of Clover Store. |  
+| 6. | Clover Developer Portal | Vue/Java/NodeJS/PostgreSQL | Rolling out Clover developer portal for the public with incentivize program |  
+| 7. | Clover Wallet Firefox/Safari Extension | Javascript/Vue | Delivering Clover Firefox/Safari extension |  
+| 8. | Clover Governance | Solidity/Rust | Releasing first build of Clover governance system |
+| 9. | Storage | NodeJS | Adding IPFS/AR/CRUST support as the Storage service |
```
Comment (laboon) [inline applications/clover_network.md:177]: These milestones seem... extraordinarily ambitious. Are you certain you can complete them in one-month time periods?

File: applications/clover_network.md (hunk 3 above)
Comment (brqgoo) [reply applications/clover_network.md:34]: Apart from basic functionalities, we’re at the moment designing some approaches on how to best implement Clover-native gassless transaction layer with our reference wallet. Design mockups will be updated accordingly.

File: applications/clover_network.md (hunk 4 above)
Comment (brqgoo) [reply applications/clover_network.md:44]: We have the protocol and Clover Store's smart contract ready for deployment. IPFS APIs will be ready by the end of January.

File: applications/clover_network.md (hunk 5 above)
Comment (brqgoo) [reply applications/clover_network.md:177]: Yes, quite ambitious to deliver such amount of work, however we have to deliver as much as we can in shortest time possible. We really need to push our limits to hit early milestones which will leave us even more milestones.

**Comment by brqgoo** [comment]
@mmagician 
1. We believe that dApp builders should benefit from making contributions to the ecosystem, and receiving part of the gas fee as their additional income. We re-parameterized the gas distribution model to direct gas fees to both network maintainers and dApp builders. This distribution happens in the scripting level within EVM module. We are also introducing a “gassless transaction layer” for relayers to act on behalf of the sender where they can cover gas price in the base currency, and receive compensation in the denominated asset. This is so that end-users can seamlessly transact their tokens by covering fees with the same token from the amount they transact, without depending on the base currency.
2. We are rolling out an enterprise pallet for managing developers virtual organisation. For example, if developers A,B and C are working together to deliver a smart contract, they can form an enterpise virtual firm which has its own address. Developers collectively deploy their contracts under their virtual firm address, this is so that all the fee income will go straight to the virtual firm.
3. We’re working on some minor adjustments like recording smart contract deployer address etc.
4. Rosetta is a protocol that contributes to 3 API sets: Data API/Construction API/Indexer API. We are utilising Rosetta Protocol to support BTC/ETH/DOT/CLV for our block explorer.

File: applications/clover_network.md (hunk 2 above)
Comment (brqgoo) [reply applications/clover_network.md]: Updated docs link accordingly: https://docs.google.com/spreadsheets/d/1BccgFYHS8YLEX5H9NpQf4ENE9Qwniq8YNpLEXXSAaRo/edit?usp=sharing

**Review by Noc2 — APPROVED** [review]

**Review by laboon — APPROVED** [review]

**Review by mmagician — APPROVED** [review]
As @laboon pointed out, the milestones are very ambitious, but good luck!

**Comment by Noc2** [comment]
Hi @brqgoo Could you please take down the announcement of the grant: https://twitter.com/clover_finance/status/1354013610024951808 See the message above: “Please don’t announce the grant publicly before you finished at least the first milestone of the project.” as well as our [badge](https://github.com/w3f/General-Grants-Program/blob/master/grants/grant-badge-guidelines.md) and [announcement guidelines](https://github.com/w3f/General-Grants-Program/blob/master/grants/announcement-guidelines.md).

**Comment by brqgoo** [comment]
Hi @Noc2 sorry for the inconvenience. We take down respective announcements on social channels.

**Comment by Noc2** [comment]
Thanks!

**Comment by brqgoo** [comment]
Hi all, we're happy to say we hit all the milestones for January 2021!

**Comment by Noc2** [comment]
@brqgoo That’s great to hear! Could you submit the milestone delivery to the following repo: https://github.com/w3f/Grant-Milestone-Delivery#mailbox-milestone-delivery-process and we will look into it as soon as possible?

**Comment by semuelle** [comment]
Contract [terminated](https://github.com/w3f/Grant-Milestone-Delivery/pull/106#issuecomment-794570629).

**Pull request merged**